contracts/citycoin.clar

```diff
@@ -620,12 +644,40 @@
 (define-public (mine-tokens (amount-ustx uint))
     (let (
         (miner-rec (get-block-miner-rec-or-default block-height))
+        (rc (unwrap! (get-reward-cycle block-height)
+            (err ERR-STACKING-NOT-AVAILABLE)))
+        (total-stacked (get total-tokens (map-get? tokens-per-cycle { reward-cycle: rc })))
+        (total-stacked-ustx (default-to u0 total-stacked))
+        (stacked-something (not (is-eq total-stacked-ustx u0)))
+        (amount-ustx-to-stacker
+            (if stacked-something
+                (/ (* SPLIT_STACKER_PERCENTAGE amount-ustx) u100)
+                u0
+            )
+        )
+        (amount-ustx-to-city
+            (if stacked-something
+                (/ (* SPLIT_CITY_PERCENTAGE amount-ustx) u100)
+                amount-ustx
+            )
+        )
     )
+
     (begin
         (try! (can-mine-tokens tx-sender block-height amount-ustx miner-rec))
 
-        (try! (set-tokens-mined tx-sender block-height amount-ustx))
-        (unwrap-panic (stx-transfer? amount-ustx tx-sender (as-contract tx-sender)))
+        (try! (set-tokens-mined tx-sender block-height amount-ustx amount-ustx-to-stacker amount-ustx-to-city))
+
+        ;; check if stacking is active
+        (if stacked-something
+            ;; transfer with split if active
+            (begin 
+                (unwrap-panic (stx-transfer? (/ (* SPLIT_STACKER_PERCENTAGE amount-ustx) u100) tx-sender (as-contract tx-sender)))
+                (unwrap-panic (stx-transfer? (/ (* SPLIT_CITY_PERCENTAGE amount-ustx) u100) tx-sender (var-get city-wallet)))
+            )
+            ;; transfer to custodied wallet if not active
+            (unwrap-panic (stx-transfer? amount-ustx tx-sender (var-get city-wallet)))
+        )
```

Comment:
I think here we can reuse variables calculated few lines above.
```suggestion
        (if stacked-something
            ;; transfer with split if active
            (begin 
                (unwrap-panic (stx-transfer? amount-ustx-to-stacker tx-sender (as-contract tx-sender)))
                (unwrap-panic (stx-transfer? amount-ustx-to-city tx-sender (var-get city-wallet)))
            )
            ;; transfer to custodied wallet if not active
            (unwrap-panic (stx-transfer? amount-ustx-to-city tx-sender (var-get city-wallet)))
        )
```

Reply:
Fixed in 29f9f6f. Also used `amount-ustx-to-city` for the 100% transfer to be consistent with using computed variables.